### PR TITLE
add apis for department with updates to Form and Department

### PR DESCRIPTION
**Pull request by Vihit**
Need to rebuild database since there is update to user table.

### CODE REVIEW

File: logever/src/main/java/ai/smartfac/logever/controller/DepartmentController.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    @Autowired
+    DepartmentService departmentService;
+    @GetMapping("/")
+    public ResponseEntity<?> getDepartments() {
+        Iterable<Department> departments = departmentService.getDepartments();
+        return new ResponseEntity<>(departments, HttpStatus.OK);
+    }
+
+    @GetMapping("/{department}/")
+    public ResponseEntity<?> getRole(@PathVariable(name = "department") String department) {
```
Comment (cpandya231) [inline logever/src/main/java/ai/smartfac/logever/controller/DepartmentController.java:28]: Rename this to getDepartmentByName

File: logever/src/main/java/ai/smartfac/logever/controller/UserController.java [hunk 2]
```diff
@@ -57,7 +57,8 @@ public ResponseEntity<?> updateUser(@RequestBody User user) {
         User updatedUser;
         boolean userUpdated = false;
         if(!existingUser.isEmpty()) {
-            if(user.getDepartment()!=null && !user.getDepartment().isEmpty() && !existingUser.get().getDepartment().equals(user.getDepartment())) {
+            System.out.println(user.getDepartment().getId());
```
Comment (cpandya231) [inline logever/src/main/java/ai/smartfac/logever/controller/UserController.java:60]: We can remove this sysout

File: logever/src/main/java/ai/smartfac/logever/controller/FormController.java [hunk 3]
```diff
@@ -24,15 +26,33 @@ public ResponseEntity<?> getForms() {
     }
 
     @GetMapping("/{form}/")
-    public ResponseEntity<?> getRole(@PathVariable(name = "form") String form) {
+    public ResponseEntity<?> getForm(@PathVariable(name = "form") String form) {
```
Comment (cpandya231) [inline logever/src/main/java/ai/smartfac/logever/controller/FormController.java:29]: Don't we need getFormById as well?

**Review by cpandya231 — CHANGES_REQUESTED** [review]

File: logever/src/main/java/ai/smartfac/logever/controller/FormController.java (hunk 3 above)
Comment (Vihit) [reply logever/src/main/java/ai/smartfac/logever/controller/FormController.java:29]: We do have form by id. This is when we click on formname from form dashboard to get the form details.

File: logever/src/main/java/ai/smartfac/logever/controller/UserController.java (hunk 2 above)
Comment (Vihit) [reply logever/src/main/java/ai/smartfac/logever/controller/UserController.java]: done

**Review by cpandya231 — APPROVED** [review]

**Pull request merged**